build/symbols.proj

```diff
@@ -15,7 +15,8 @@
   <Target Name="GetSymbolsAndAssembliesToIndex">
     <MSBuild
       Projects="@(SolutionProjectsWithoutVSIX)"
-      Targets="GetSymbolsToIndex">
+      Targets="GetSymbolsToIndex"
+      Properties="NoBuild=true;">
```

Comment:
May I know what's this for? Thanks.

Reply:
If you run `symbols.proj` locally, you'll see that it re-runs ilmerge on nuget.exe, and probably the NuGet.Build.Tasks.Pack.dll. This means, the exe, and symbols for these components are regenerated. I don't know what heuristics the debugger uses to determine the symbols "key" to get from the server, but if the second ilmerge changes the exe and pdb too much, then the symbols we publish will not match the binaries published, and noone will be able to debug those binaries.

So, by telling the .NET SDK not to rebuild, it avoids our ilmerge target from running, and we can publish the pdbs that were generated at the same time the published binaries were generated.

An alternative, which we should eventually do anyway, is improve our ilmerge task to use [msbuild's incremental build](https://docs.microsoft.com/en-us/visualstudio/msbuild/incremental-builds?view=vs-2019) to avoid needlessly ilmerging so many times. When running builds locally, or if you look carefully at our CI logs, you'll see we actually ilmerge nuget.exe multiple times, because msbuild doesn't know how to determine if the previous copy is already up-to-date.  THis is a particular risk for signed assemblies, because if we ilmerege a new nuget.exe and overwrite a signed nuget.exe, then we lose the authenticode signature.